### PR TITLE
[images] new image build acs-engine and upload

### PR DESCRIPTION
**Pull request by papagalu**
-also changed the acs_engine_prow_test to parse the junit artifacts
     and fail if any other tast case except "Test" fails

### CODE REVIEW

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **taeduard**

Assign the PR to them by writing `/assign @taeduard` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/e2e-win-prow-deployment/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: images/acs_engine_build/build_acs.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+ACS_DIR=${GOPATH}/src/github.com/Azure/acs-engine
+ACS_BIN="$ACS_DIR/bin/acs-engine"
+
+OUTPUT_DIR=$HOME/output
+ARTIFACTS_DIR=$OUTPUT_DIR/artifacts
+OUTPUT_FILE=${OUTPUT_DIR}/build-log.txt
+
+KUBETEST_STORAGE_CONTAINER="k8s-windows"
+KUBETEST_STORAGE_BLOB="testing/kubetest/kubetest_$(date '+%Y-%m-%d-%H-%M-%S')/kubetest"
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: What's kubetest got to do with anything?

File: images/acs_engine_build/build_acs.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+# Global vars
+REPO=${REPO:-"http://github.com/Azure/acs-engine"}
+BRANCH=${BRANCH:-"master"}
+
+ACS_DIR=${GOPATH}/src/github.com/Azure/acs-engine
+ACS_BIN="$ACS_DIR/bin/acs-engine"
+
+OUTPUT_DIR=$HOME/output
+ARTIFACTS_DIR=$OUTPUT_DIR/artifacts
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: Why do we need this?

File: images/acs_engine_build/build_acs.sh [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+OUTPUT_FILE=${OUTPUT_DIR}/build-log.txt
+
+KUBETEST_STORAGE_CONTAINER="k8s-windows"
+KUBETEST_STORAGE_BLOB="testing/kubetest/kubetest_$(date '+%Y-%m-%d-%H-%M-%S')/kubetest"
+KUBETEST_STORAGE_BLOB_LATEST="testing/kubetest/kubetest_latest/kubetest"
+
+mkdir -p ${OUTPUT_DIR}
+mkdir -p ${ARTIFACTS_DIR}
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: ditto

File: images/acs_engine_build/build_acs.sh [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+popd () {
+    command popd "$@" > /dev/null
+}
+
+function upload_results {
+    echo "Uploading results"
+    gcloud auth activate-service-account --key-file=${GOOGLE_APPLICATION_CREDENTIALS}
+    gsutil cp -r ${OUTPUT_DIR} ${GS_BUCKET_FULL_PATH}
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: Who is GS_BUCKET_FULL_PATH ?

File: images/acs_engine_build/build_acs.sh [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+function upload_blob() {
+    az storage blob upload --container-name $1 --name $2 --file $BAZEL_OUTPUT --account-name $AZ_STORAGE_ACCOUNT --account-key $AZ_STORAGE_KEY
+}
+
+function main () {
+    build_acs_engine
+
+    echo "Uploading kubetest to storage blob"
+    upload_blob $KUBETEST_STORAGE_CONTAINER $KUBETEST_STORAGE_BLOB
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: This is just lazy copy/paste... again, what's kubetest got to do with this?

File: images/acs_engine_build/build_acs.sh [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+        # build acs-engine
+        echo "Installing dependencies"
+        glide install || true
+        echo "Building acs-engine"
+        make build
+    popd
+}
+
+function upload_blob() {
+    az storage blob upload --container-name $1 --name $2 --file $BAZEL_OUTPUT --account-name $AZ_STORAGE_ACCOUNT --account-key $AZ_STORAGE_KEY
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: Who is BAZEL_OUTPUT ?

File: images/acs_engine_build/build_acs.sh [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+function main () {
+    build_acs_engine
+
+    echo "Uploading kubetest to storage blob"
+    upload_blob $KUBETEST_STORAGE_CONTAINER $KUBETEST_STORAGE_BLOB
+    upload_blob $KUBETEST_STORAGE_CONTAINER $KUBETEST_STORAGE_BLOB_LATEST
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: Ditto

File: images/acs_engine_build/build_acs.sh [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+            git merge --no-ff --m "PR to test #${PULL_NUMBER}" pr/$PULL_NUMBER
+        fi
+
+        go get github.com/Masterminds/glide
+        
+        # build acs-engine
+        echo "Installing dependencies"
+        glide install || true
+        echo "Building acs-engine"
+        make build
```
Comment (adelina-t) [inline images/acs_engine_build/build_acs.sh]: And you take the build output and put it in a tar.gz where ?

File: images/acs_engine_build/build_acs.sh (hunk 2 above)
Comment (papagalu) [reply images/acs_engine_build/build_acs.sh]: we need it for the junit_runner.xml, so we can parse it

Comment (adelina-t) [reply images/acs_engine_build/build_acs.sh]: really? and where do you parse it? Who exactly outputs junit_runner.xml if you're not running kubetest inside this job?

Comment (adelina-t) [reply images/acs_engine_build/build_acs.sh]: This is the script for the job that builds and uploads acs-engine right?

Comment (papagalu) [reply images/acs_engine_build/build_acs.sh]: yeah, though initially that it was in the other script, my bad

**Pull request merged**